### PR TITLE
Security fixes according to Brakeman

### PR DESCRIPTION
**Pull request by vpetrusenko**
Related to issue #336

### CODE REVIEW

File: app/controllers/test/module_controller.rb [hunk 1]
```diff
@@ -1,5 +1,7 @@
 module Test
   class ModuleController < ActionController::Base
+    protect_from_forgery with: :exception
```
Comment (yivo) [inline app/controllers/test/module_controller.rb]: @vpetrusenko Do you know what this controller is used for any why it doesn't require such stuff?

Comment (yivo) [reply app/controllers/test/module_controller.rb]: You need to remove it.

**Review by yivo — CHANGES_REQUESTED** [review]

File: app/controllers/test/module_controller.rb (hunk 1 above)
Comment (vpetrusenko) [reply app/controllers/test/module_controller.rb]: @yivo sorry, i just follow brakeman advices, will be carefully

Comment (yivo) [reply app/controllers/test/module_controller.rb]: @vpetrusenko Don't respect too much from automated tool for analysing static code. Will you continue fixing brakeman reports in this PR or create new one?

**Review by yivo — CHANGES_REQUESTED** [review]

File: app/controllers/test/module_controller.rb (hunk 1 above)
Comment (vpetrusenko) [reply app/controllers/test/module_controller.rb]: @yivo i will continue in this PR

**Comment by yivo** [comment]
Hello @vpetrusenko. What is the progress on it?

**Comment by vpetrusenko** [comment]
@yivo Hello, sorry i don't work on it in because of barong rush, but i work on it today

**Comment by yivo** [comment]
@vpetrusenko I have checked PR and noticed you are doing many things wrong. Consider digging into the implementation details to understand hot things work. Remember it is STATIC code analysis. It is mostly stupid and fake-positive in 90% cases.

**Comment by vpetrusenko** [comment]
@yivo okay i will try to understand and fix my mistakes, but actually i don't understand a lot of brakeman remarks about peatio

**Comment by vpetrusenko** [comment]
@yivo So all other brakeman remarks for me seems unsolvable in case to keep peatio work correct, so if this changes looks good i guess we should leave only them in patch

File: app/models/account_version.rb [hunk 2]
```diff
@@ -49,7 +49,7 @@ def self.optimistically_lock_account_and_create!(balance, locked, attrs)
     values       = ActiveRecord::Base.send :sanitize_sql_array, values_array
 
     select = Account.unscoped.select(values).where(id: account_id, balance: balance, locked: locked).to_sql
-    stmt   = "INSERT INTO account_versions (#{attrs.keys.join(',')}) #{select}"
+    stmt   = sanitize_sql("INSERT INTO account_versions (#{attrs.keys.join(',')}) #{select}")
```
Comment (yivo) [inline app/models/account_version.rb]: @vpetrusenko I am very sorry but WTF? Have you read the source of `sanitize_sql`? Do you understand it?

File: app/views/private/assets/_usd_assets.html.erb [hunk 3]
```diff
@@ -21,7 +21,7 @@
           <div class="clearfix"></div>
         </div>
 
-        <%= t('.description', @usd_assets['accounts'].first.symbolize_keys).html_safe %>
+        <%= t('.description', @usd_assets['accounts'].first.symbolize_keys.html_safe) %>
```
Comment (yivo) [inline app/views/private/assets/_usd_assets.html.erb]: @vpetrusenko Why you haven't run the project?
screen shot 2018-02-02 at 9 45 09 pm

File: app/controllers/application_controller.rb [hunk 4]
```diff
@@ -18,7 +18,7 @@ def current_market
 
   def redirect_back_or_settings_page
     if cookies[:redirect_to].present?
-      redirect_to cookies[:redirect_to]
+      redirect_to URI.parse(cookies[:redirect_to]).path
```
Comment (yivo) [inline app/controllers/application_controller.rb:21]: @vpetrusenko OK.

**Review by yivo — COMMENTED** [review]
@vpetrusenko I will do changes by myself. Please DON'T push to this branch. Thanks.

**Comment by yivo** [comment]
@vpetrusenko I have pushed the changes. I will wait until all specs will success.

**Comment by yivo** [comment]
It works well: deposits / withdraws / trading.

**Review by yivo — APPROVED** [review]

**Review by mod — APPROVED** [review]

**Comment by vpetrusenko** [comment]
@yivo thank you for help with this patch!

**Pull request merged**